tests/unit/spec/base/mozilla-utils.js

```diff
@@ -9,6 +9,67 @@ describe('mozilla-utils.js', function() {
 
     'use strict';
 
+    describe('trans', function () {
+        var stringDiv;
+
+        beforeEach(function () {
+            stringDiv = '<div id="strings" data-global-close="Close" ' +
+            'data-global-next="Next" ' +
+            'data-global-previous="Previous"> ' +
+            '</div>';
+
+            var container = document.createElement('div');
+            container.innerHTML = stringDiv;
+            $(container).appendTo('body');
+        });
+
+        afterEach(function() {
+            $(stringDiv).remove();
+        });
+
+        it('should correctly return translation value', function () {
+            var translation = Mozilla.Utils.trans('global-next');
+            expect(translation === 'Next');
+        });
+    });
+
+    describe('initLangSwitcher', function () {
+        var form;
+
+        beforeEach(function () {
+            form = '<form id="lang_form" method="get" action="#">' +
+              '<label for="page-language-select">Langue</label>' +
+              '<select id="page-language-select" name="lang" dir="ltr">' +
+                  '<option lang="de" value="de">Deutsch</option>' +
+                  '<option lang="en-US" value="en-US" selected>English</option>' +
+                  '<option lang="fr" value="fr">Français</option>' +
+              '</select>' +
+              '<button type="submit">Go</button>' +
+            '</form>';
+
+            var container = document.createElement('div');
+            container.innerHTML = form;
+            $(container).appendTo('body');
+        });
+
+        afterEach(function() {
+            $(form).remove();
+        });
+
+        it('should correctly identify the previous and new languages', function () {
+            spyOn(Mozilla.Utils, 'doRedirect');
+            var $select = $('#page-language-select');
+            // init
+            Mozilla.Utils.initLangSwitcher();
+            // change the drop down
+            $select.val('de');
+            // fire the event
+            $select.trigger('change');
+            // check redirected to expected page
+            expect(window.location.search === '?lang=de#&step=1');
```

Comment:
What I find slightly uneasy is we're already depending on the output of a completely unrelated tests here in order for this particular test to pass. From what I can see, `&step=1` is an unintended side effect from an existing test, so if that changes then this test suddenly starts failing. Ideally, I'd like to not have the test runner put through any kind of real redirects at all, as the more things like this we add the more prone it becomes to odd errors that are hard to debug :/

Reply:
Updated just to look for the `lang=de` value.

Though I suppose this will be deleted in #6635